flambe/experiment/experiment.py

```diff
@@ -274,8 +274,12 @@ def run(self, force: bool = False, verbose: bool = False, debug: bool = False, *
 
         # By default use all CPUs if no GPU is present
         devices = self.devices if self.devices else None
-        if devices is None and utils.local_has_gpu():
-            devices = {"cpu": 4, "gpu": 1}
+        if devices is None:
+            cluster_devices = ray.cluster_resources()
+            if 'GPU' in cluster_devices or 'gpu' in cluster_devices:
+                devices = {"cpu": 4, "gpu": 1}
```

Comment:
Does this cover all cases? i.e. do we not support parallelizing variants on a single multi-gpu machine?

Reply:
we do, this is exactly what this PR addresses. This is only for when devices is not provided as a sensible default, otherwise the user should specify